test/QCheck/DataProvider.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace QCheck;
+
+class DataProviderTest extends \PHPUnit_Framework_TestCase {
+    function provider($name, $n = 10) {
+        return DataProvider::provider(array(
+            Generator::strings(),
+            Generator::ints(),
+            Generator::booleans()
+        ), $n);
+    }
+
+    /**
+     * @dataProvider provider
+     */
+    function testDataProvider($s, $i, $b) {
+        // this test is only supposed to prove that we have a valid provider
+        $this->assertTrue(is_string($s));
```

Comment:
Using `$this->assertInternalType('string', $s);` is preferred over using `assertTrue`